yearn/contracts/SaddleStrategy.sol

```diff
@@ -0,0 +1,395 @@
+    using SafeERC20 for IERC20;
+    using Address for address;
+    using SafeMath for uint256;
+
+    // Address of the KEEP token contract.
+    address public constant keep = 0x85Eee30c52B0b379b046Fb0F85F4f3Dc3009aFEC;
+    // Address of the WETH token contract.
+    address public constant weth = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+    // Address of the WBTC token contract.
+    address public constant wbtc = 0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599;
+    // Address of the Uniswap V2 router contract.
+    address public constant uniswap =
+        0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D;
+    // Address of the Saddle Swap contract, taken from
+    // https://docs.saddle.finance/contracts
+    address public constant swap = 0x4f6A43Ad7cba042606dECaCA730d4CE0A57ac62e;
+    // Address of the tBTC v2 Saddle reward pool.
+    address public lpRewardsTbtcV2Saddle;
+
+    constructor(address _vault, address _lpRewardsTbtcV2Saddle)
+        public
+        BaseStrategy(_vault)
+    {
+        // TODO: Check what the correct values should be
+        // Strategy settings.
+        minReportDelay = 6 hours;
+        maxReportDelay = 2 days;
+        profitFactor = 1;
+        debtThreshold = 1e24;
+
+        // tBTC-related settings.
+        lpRewardsTbtcV2Saddle = _lpRewardsTbtcV2Saddle;
+        address lpToken = ILPRewardsTBTCV2Saddle(_lpRewardsTbtcV2Saddle)
+        .wrappedToken();
+        require(lpToken == address(want), "Incorrect reward pool LP token");
+    }
+
+    /// @return Name of the Yearn vault strategy.
+    function name() external view override returns (string memory) {
+        return
+            string(
+                abi.encodePacked(
+                    "Saddle",
+                    IERC20Metadata(address(want)).symbol()
+                )
+            );
+    }
+
+    /// @return Balance of the vault's underlying token under management.
+    function balanceOfWant() public view returns (uint256) {
+        return want.balanceOf(address(this));
+    }
+
+    /// @return Balance of the vault's underlying token staked into the tBTC v2
+    ///         Saddle reward pool.
+    function balanceOfPool() public view returns (uint256) {
+        return
+            ILPRewardsTBTCV2Saddle(lpRewardsTbtcV2Saddle).balanceOf(
+                address(this)
+            );
+    }
+
+    /// @return Sum of balanceOfWant and balanceOfPool.
+    function estimatedTotalAssets() public view override returns (uint256) {
+        return balanceOfWant().add(balanceOfPool());
+    }
+
+    /// @notice This method is defined in the BaseStrategy contract and is meant
+    ///         to provide an accurate conversion from amtInWei (denominated in wei)
+    ///         to want token (using the native decimal characteristics of want token).
+    /// @param amtInWei The amount (in wei/1e-18 ETH) to convert to want tokens.
+    /// @return The amount in want tokens.
+    function ethToWant(uint256 amtInWei)
```

Comment:
Nitpick: Let's move it at the bottom of the contract as we do in other strategies. This method is not strictly related to strategy cashflow.

Reply:
Done in https://github.com/keep-network/tbtc-v2/pull/13/commits/227ed9911ad590b61480bb840375083958d2233b.